python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -367,6 +367,31 @@ def get_current_timestamp(self, current_time: Optional[datetime] = None) -> floa
             else pendulum.now(self.timezone)
         ).timestamp()
 
+    def _get_fast_num_partitions(self, current_time: Optional[datetime] = None) -> Optional[int]:
+        last_partition_window = self.get_last_partition_window(current_time)
+        first_partition_window = self.get_first_partition_window(current_time)
+
+        if not last_partition_window or not first_partition_window:
+            return None
+
+        if self.is_basic_daily:
+            return (last_partition_window.start - first_partition_window.start).days + 1
+
+        fixed_minute_interval = self.get_fixed_minute_interval()
+        if fixed_minute_interval:
+            return (
+                int(
```

Comment:
Feel free to disregard, but I think this would read more clearly as something like:
```python
minutes_in_window = (last_partition_window.start.timestamp() - first_partition_window.start.timestamp()) / 60
return minutes_in_window // fixed_minute_interval + 1
```